ras/perf: Add perf tool testcase

kamalesh-babulal opened the pull request:
This testcase verifies the following perf subcommands:
- help
- version
- list
- record
- report
- kallsyms
- annotate
- evlist
- stat
- bench

Additionally executes the builtin test benchmark

Signed-off-by: Kamalesh Babulal <kamalesh@linux.vnet.ibm.com>

kamalesh-babulal commented:
@nasastry Thanks for the review.

kamalesh-babulal commented:
v2 Changes:
- Added the `SuSE` to the `RHEL` distro
- Explicitly add outfile to `perf record`
- Added `perf script` command
- Improved the subcommand execution logic

harish-24 commented on ras/perf.py:
```diff
@@ -0,0 +1,129 @@
+from avocado.utils.software_manager import SoftwareManager
+
+
+class perf(Test):
```
Class name `Perf` would be appropriate here.

harish-24 commented on ras/perf.py:
```diff
@@ -0,0 +1,129 @@
+    fail_cmd = list()
+
+    def run_cmd(self, cmd, verbose=True):
+        self.log.info("executing ============== %s =================" % cmd)
```
Minor suggestion. `self.log.info` is usually used with `,`.
```        self.log.info("executing ============== %s =================", cmd)```
Just a nit pick, can be left as is.

harish-24 commented on ras/perf.py:
```diff
@@ -0,0 +1,129 @@
+            self.is_fail += 1
+            self.fail_cmd.append(cmd)
+
+    def run_cmd_out(self, cmd):
```
As it is a generic function(not specific to class), better to add `@staticmethod` before the method. Refer https://github.com/avocado-framework-tests/avocado-misc-tests/blob/master/cpu/cpustress.py#L80 for the same.

harish-24 commented on ras/perf.py:
```diff
@@ -0,0 +1,129 @@
+            if not smg.check_installed(pkg) and not smg.install(pkg):
+                self.cancel("Package %s is missing/could not be installed" % pkg)
+
+    def testPerfShort(self):
```
minor suggestion: better to have `test_perf_short`. Camel case is usually used for classes. Same for other `test*` methods.

kamalesh-babulal commented:
@harish-24 Thanks for the review. 

**V3 Changes:**
- Added `@staticmethod` for `run_cmd_out()`
- Changed the function naming convention to c function style.
- `self.log.info` replaced `%` with `,`

PraveenPenguin commented:
@nasastry @harish-24 thanks for review
